FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/DemoTasks/simple_sub_pub_demo.c

```diff
@@ -0,0 +1,546 @@
+    xPublishInfo.pPayload = payloadBuf;
+
+    /* Store the handler to this task in the command context so the callback
+     * that executes when the command is acknowledged can send a notification
+     * back to this task. */
+    memset( ( void * ) &xCommandContext, 0x00, sizeof( xCommandContext ) );
+    xCommandContext.xTaskToNotify = xTaskGetCurrentTaskHandle();
+
+    xCommandParams.blockTimeMs = mqttexampleMAX_COMMAND_SEND_BLOCK_TIME_MS;
+    xCommandParams.cmdCompleteCallback = prvPublishCommandCallback;
+    xCommandParams.pCmdCompleteCallbackContext = &xCommandContext;
+
+    /* For a finite number of publishes... */
+    for( ulValueToNotify = 0UL; ulValueToNotify < mqttexamplePUBLISH_COUNT; ulValueToNotify++ )
+    {
+        /* Create a payload to send with the publish message.  This contains
+         * the task name and an incrementing number. */
+        snprintf( payloadBuf,
+                  mqttexampleSTRING_BUFFER_LENGTH,
+                  "%s publishing message %d",
+                  taskName,
+                  ( int ) ulValueToNotify );
+
+        xPublishInfo.payloadLength = ( uint16_t ) strlen( payloadBuf );
+
+        /* Also store the incrementing number in the command context so it can
+         * be accessed by the callback that executes when the publish operation
+         * is acknowledged. */
+        xCommandContext.ulNotificationValue = ulValueToNotify;
+
+        LogInfo( ( "Sending publish request to agent with message \"%s\" on topic \"%s\"",
+                   payloadBuf,
+                   pcTopicBuffer ) );
+
+        /* To ensure ulNotification doesn't accidentally hold the expected value
+         * as it is to be checked against the value sent from the callback.. */
+        ulNotification = ~ulValueToNotify;
+
+        xCommandAdded = MQTTAgent_Publish( &xGlobalMqttAgentContext,
+                                           &xPublishInfo,
+                                           &xCommandParams );
+        configASSERT( xCommandAdded == MQTTSuccess );
+
+        /* For QoS 1 and 2, wait for the publish acknowledgment.  For QoS0,
+         * wait for the publish to be sent. */
+        LogInfo( ( "Task %s waiting for publish %d to complete.",
+                   taskName,
+                   ulValueToNotify ) );
+        prvWaitForCommandAcknowledgment( &ulNotification );
+
+        /* The value received by the callback that executed when the publish was
+         * acked came from the context passed into MQTTAgent_Publish() above, so
+         * should match the value set in the context above.  However QoS 0 does
+         * not provide guaranteed delivery so it is ok for the values not to match
+         * if xQos is 0.*/
+        configASSERT( ( ulNotification == ulValueToNotify ) || ( xQoS == 0 ) );
+
+        if( ulNotification == ulValueToNotify )
+        {
+            LogInfo( ( "Received ack from publishing to topic %s.",
```

Comment:
The use of "ack" may be confusing for a reader. Should it say something like..
```suggestion
            LogInfo( ( "Received notification about completion of publish operation: Topic=%s.",
```